script/testing/junit/src/TestUtility.java

```diff
@@ -9,30 +10,54 @@
 import java.sql.SQLException;
 import java.util.Properties;
 import static org.junit.Assert.assertEquals;
+import java.io.File;
+import java.io.IOException;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.w3c.dom.Element;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
 
 public class TestUtility {
     public static Connection makeDefaultConnection() throws SQLException {
         return makeConnection("localhost", 15721, "terrier");
     }
 
     public static Connection makeConnection(String host, int port, String username) throws SQLException {
-        String url = String.format("jdbc:postgresql://%s:%d/", host, port);
         Properties props = new Properties();
         props.setProperty("user", username);
-        props.setProperty("preferQueryMode", "simple"); // force SimpleQuery protocol for now
         props.setProperty("prepareThreshold", "0"); // suppress switchover to binary protocol
+
+        try {
+            File optionsFile = new File("./out/options.xml");
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document document = builder.parse(optionsFile);
+            Element options = document.getDocumentElement();
+
+            // read the preferQueryMode from the options.xml
+            String preferQueryMode = (String) options.getElementsByTagName("QueryMode").item(0).getTextContent();
+            if (!preferQueryMode.isEmpty()) {
+                props.setProperty("preferQueryMode", preferQueryMode);
+            }
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            e.printStackTrace();
```

Comment:
Should we exit or something if an exception happens?

Reply:
the configs in the `options.xml` are optional, and it is ok to leave the `preferQueryMode` blank as Andy last time mentioned